quickwit-search/src/client.rs

```
@@ -149,47 +149,62 @@ impl SearchServiceClient {
-                            result_sender.send(search_result).map_err(|_| {
-                                SearchError::InternalError(
-                                    "Sender closed, could not send leaf result.".into(),
-                                )
-                            })?;
+                let mut tonic_request = Request::new(request);
+                global::get_text_map_propagator(|propagator| {
+                    propagator.inject_context(
+                        &tracing::Span::current().context(),
+                        &mut MetadataMap(tonic_request.metadata_mut()),
+                    )
+                });
+                let tonic_result = grpc_client_clone
+                    .leaf_search_stream(tonic_request)
+                    .await
+                    .map_err(|tonic_error| parse_grpc_error(&tonic_error));
+                let (result_sender, result_receiver) = tokio::sync::mpsc::unbounded_channel();
+                // Propagate the tonic error with the channel.
+                // This may sound quirky but this makes errors handling easier when you want to
+                // retry on a stream.
+                if tonic_result.is_err() {
+                    // Receiver cannot be closed here, ignore error.
+                    let _ = result_sender.send(Err(tonic_result.unwrap_err()));
+                } else {
+                    tokio::spawn(
+                        async move {
+                            let mut results_stream = tonic_result
+                                .unwrap()
+                                .into_inner()
+                                .map_err(|tonic_error| parse_grpc_error(&tonic_error));
+                            while let Some(search_result) = results_stream.next().await {
+                                result_sender.send(search_result).map_err(|_| {
+                                    SearchError::InternalError(
```

What is the point of this error conversion?
The error is actually silently dropped.

---

yes this is useless